Server/urlService/index.js

```diff
@@ -0,0 +1,78 @@
+const database = require('../database');
+
+// Functions
+function intToBase36(x) {
+    return x.toString(36);
+}
+
+exports.findURL = async function (ctx, miniURL) {
```

Comment:
Ideally `urlService` wouldn't have to know anything about `ctx`, and would instead throw its own (custom?) errors, which would be caught in the route handler, adjusted as necessary and passed to `ctx.throw()`, but for the purposes of this exercise I don't think we need to get that involved. But just as an example (I haven't actually run this; might not be perfect working code)...

```
// Throw error if query returns nothing
if (query.rows[0] == null) {
  throw new NoSuchRecordError(`no record found for ${miniUrL}`);
}

// ...and in handler...

.get('/t/:miniURL', async (ctx) => {
  // find url in DB or error- does not exist
  try {
    const url = await urlService.findURL(ctx, ctx.params.miniURL);
    ctx.redirect(url);
  } catch (err) {
    if (err instanceof NoSuchRecordError) {
      return ctx.throw(404);
    }

    ctx.throw();
  }
})
```